examples/cdk-examples-typescript/resource-overrides/index.ts

```diff
@@ -0,0 +1,100 @@
+        bucketResource2.addPropertyDeletionOverride('BucketEncryption.ServerSideEncryptionConfiguration.0.ServerSideEncryptionByDefault');
+
+        return;
+
+        //
+        // Accessing the L1 bucket resource from an L2 bucket
+        //
+
+        const bucketResource = bucket.findChild('Resource') as s3.cloudformation.BucketResource;
+        const anotherWay = bucket.children.find(c => (c as cdk.Resource).resourceType === 'AWS::S3::Bucket') as s3.cloudformation.BucketResource;
+        assert.equal(bucketResource, anotherWay);
+
+        //
+        // This is how to specify resource options such as dependencies, metadata, update policy
+        //
+
+        bucketResource.addDependency(otherBucket.findChild('Resource') as cdk.Resource);
+        bucketResource.options.metadata = { MetadataKey: 'MetadataValue' };
+        bucketResource.options.updatePolicy = {
+            autoScalingRollingUpdate: {
+                pauseTime: '390'
+            }
+        };
+
+        //
+        // This is how to specify "raw" overrides at the __resource__ level
+        //
+
+        bucketResource.addOverride('Type', 'AWS::S3::Bucketeer'); // even "Type" can be overridden
+        bucketResource.addOverride('Transform', 'Boom');
+        bucketResource.addOverride('Properties.CorsConfiguration', {
+            Custom: 123,
+            Bar: [ 'A', 'B' ]
+        });
+
+        // addPropertyOverrides simply allows you to omit the "Properties." prefix
+        bucketResource.addPropertyOverride('VersioningConfiguration.Status', 'NewStatus');
+        bucketResource.addPropertyOverride('Foo', null);
+        bucketResource.addPropertyOverride('Token', otherBucket.bucketArn); // use tokens
+        bucketResource.addPropertyOverride('LoggingConfiguration.DestinationBucketName', otherBucket.bucketName);
+
+        //
+        // It is also possible to request a deletion of a value by either assigning
+        // `undefined` (in supported languages) or use the `addDeletionOverride` method
+        //
+
+        bucketResource.addDeletionOverride('Metadata');
+        bucketResource.addPropertyDeletionOverride('CorsConfiguration.Bar');
+
+        //
+        // It is also possible to specify overrides via a strong-typed property
+        // bag called `propertyOverrides`
+        //
+
+        bucketResource.propertyOverrides.analyticsConfigurations = [
+            {
+                id: 'config1',
+                storageClassAnalysis: {
+                    dataExport: {
+                        outputSchemaVersion: '1',
+                        destination: {
+                            format: 'html',
+                            bucketArn: otherBucket.bucketArn // use tokens freely
+                        }
+                    }
+                }
+            }
+        ];
+
+        bucketResource.propertyOverrides.corsConfiguration = {
```

Comment:
The names in this section don't fully convey what is happening for me. I'm not sure I can do better. The `raw` overrides to me seem like `synthOverride` or `synthPropertyOverride` and the typed overrides seem like singular things `subnetResource.propertyOverride.mapPublicIpOnLaunch = true;` Even in the situation where the property is an object or list, such as the example the override is for the singular object `analyticsConfigurations`.